### PR TITLE
Add support for origin registration completeness status

### LINKED ISSUE #1130: Web UI didn't redirect to `redirectURL` once user logged in
Pelican Version:
_Run `./pelican --version` to get version number._

Pelican Service:
- [x] Registry
- [x] Director
- [x] Origin
- [x] Cache

**Describe the bug**
_A clear and concise description of what the bug is._

If a user hit s page where requires login, the user will be taken to the login page. If the login is successful, the user should be redirect back to the original page. Current this doesn't work. The user was only redirected to the root page at `/view/`

**To Reproduce**
_Steps to reproduce the behavior._

**Expected behavior**
_A clear and concise description of what you expected to happen._

**Screenshots**
_If applicable, add screenshots to help explain your problem._

**Desktop (please complete the following information):**
 - OS: [e.g. Linux, MacOS]
 - Browser: [e.g. chrome, safari, if the bug relates to any web user interface]

**Additional context**
_Add any other context about the problem here._

### PR DESCRIPTION
**Pull request by haoming29**
Related issue: #1105. This PR implements the API for enabling the feature mentioned in the ticket, but the UI components is not in the scope of this PR.
Also fixes #1130 

Some noticeable changes this PR made:
* Allow user with a valid access token to view and edit a namespace registration in the registry that has an empty `UserID` field. A valid access token is a JWT signed by the origin with the private key matches public key from the the namespace registration to access
* Origin UI API `/exports` now has three additional fields: `edit_url`, `status`, `status_description` as documented in the updated Swagger spec. `status` shows the registration status of the namespace, and `status_description` provides a detailed description of the status. `edit_url` will take the user from the origin to the corresponding registry page to edit the registration with a valid access token embedded.
* Streamlined web UI redirects with the following updated policies:
  * For an unauthenticated request hit an auth required page, the redirect to `/login` page is handled by the **frontend**, and frontend will "save" the URL to the current page as a query parameter `redirectURL` and add it to the `/login` URL.
  * When a user is logged in, either via admin password (`/login` page) or oauth callback (`/auth/oauth/callback`), the redirect to the original page was handled by the **backend**.

### CODE REVIEW

File: registry/registry_ui.go
Comment (jhiemstrawisc) [inline registry/registry_ui.go:523]: Is this actually a "namespace not found" error, or is it a "permission denied" error? The message seems to suggest both.

File: registry/registry_ui.go
Comment (jhiemstrawisc) [inline registry/registry_ui.go:571]: Should these logged errors also enforce a lowercase? Or do we only care about then when passing errors back up the error stack?

File: swagger/pelican-swagger.yaml
Comment (jhiemstrawisc) [inline swagger/pelican-swagger.yaml:425]: How is "Not Supported" different from "Registration Error"?

File: web_ui/authentication_test.go
Comment (jhiemstrawisc) [inline web_ui/authentication_test.go:302]: Good catch :)

File: web_ui/ui.go
Comment (jhiemstrawisc) [inline web_ui/ui.go:189]: Is the trailing `/` before the query necessary? Usually I see queries take the form `/some/path?query` and not `/some/path/?query`

File: web_ui/ui.go
Comment (jhiemstrawisc) [inline web_ui/ui.go:197]: See previous comment

**Review by jhiemstrawisc — CHANGES_REQUESTED** [review]
A few minor changes, and a note that I skimmed any changes under web_ui (not something I know enough about to review).

File: registry/registry_ui.go
Comment (haoming29) [reply registry/registry_ui.go:523]: This is the part I'm not very confident about. I saw people returning 404 for objects that does not belong to a user (although it exists), (to confuse attackers?), but this will also somehow confuse our users a bit.

File: registry/registry_ui.go
Comment (haoming29) [reply registry/registry_ui.go:571]: The logged error are not chained together (unless the real Go error), so I don't think we should put them in lower case if they are meant to be a complete sentence.

File: swagger/pelican-swagger.yaml
Comment (haoming29) [reply swagger/pelican-swagger.yaml:425]: "Not Support" is when the registry is below 7.8.0 (or whichever version this PR is included in)
"Registration Error" means that there's an issue with the registration itself, like not having identity when registering a namespace overlapped with topology, or something else that leads to a failure of registering the namespace. In this case, the "completeness status" doesn't make any sense because the namespace is not registered at all.

Comment (haoming29) [reply swagger/pelican-swagger.yaml:425]: I will update the Swagger doc to detail them

File: web_ui/ui.go
Comment (haoming29) [reply web_ui/ui.go:189]: Yeah that's our set convention, we enforce the trailing slash for frontend routes

File: registry/registry_ui.go
Comment (jhiemstrawisc) [reply registry/registry_ui.go:523]: I think we're probably safe to expose its under-construction existence here, especially if there are any other queries an attacker could perform to learn about it. Presumably, namespaces are public once they're registered anyway, so existence doesn't seem like it needs to be tightly protected.

**Review by jhiemstrawisc — APPROVED** [review]
LGTM! I'll approve and let you merge when you're ready -- I think we're still discussing one of the comments, but you get final say.

**Comment by haoming29** [comment]
@jhiemstrawisc I agree with you, and for the sake of consistency, I updated the status code to use 403 and error messages for the `getNamespace` handler as well.

**Pull request merged**
